circeSupport/src/main/scala/io.buildo.enumero/CirceSupport.scala

```diff
@@ -15,21 +16,37 @@ trait CirceSupport {
 
   implicit def decodeCaseEnum[E <: CaseEnum](implicit s: CaseEnumSerialization[E]): Decoder[E] =
     Decoder.decodeString.emap { str =>
-      s.caseFromString(str).map(Right(_)).getOrElse(Left(deserializationError(str, s.name)))
+      s.caseFromString(str)
+        .map(Right(_))
+        .getOrElse(Left(deserializationError(str, s.name)))
     }
 
+  implicit final def decodeKeyCaseEnum[E <: CaseEnum](
```

Comment:
lines 24 to 33 + minor changes in `build.sbt` + new tests in `CirceSupportSpec.scala` are the only things that you have to review @bytecodeguru  (other changes are just `scalafmt` formatting)